src/Generator/CreditMemoGenerator.php

```diff
@@ -37,15 +38,15 @@ final class CreditMemoGenerator implements CreditMemoGeneratorInterface
     private $uuidCreditMemoIdentifierGenerator;
 
     public function __construct(
-        CreditMemoUnitGeneratorInterface $orderItemUnitCreditMemoUnitGenerator,
-        CreditMemoUnitGeneratorInterface $shipmentCreditMemoUnitGenerator,
+        LineItemsConverterInterface $lineItemsConverter,
+        LineItemsConverterInterface $shipmentLineItemsConverter,
```

Comment:
As the order of these services is important, I would recommend creating two interfaces instead of one. Otherwise, errors may be really misleading (when `units` would be passed to `shipmentLineItemsConvertera` and `shipments` to `lineItemsConverter`)

Reply:
I'm not convinced to that, the input and output are the same and having the same interface is logical to me

Reply:
Maybe the best would be the queue with priorities accepting only array of specific units 💃 But I would leave it as it is right now